compute_tools/src/spec.rs

```diff
@@ -152,8 +152,12 @@ pub fn handle_roles(spec: &ComputeSpec, client: &mut Client) -> Result<()> {
             {
                 RoleAction::Update
             } else if let Some(pg_pwd) = &r.encrypted_password {
-                // Check whether password changed or not (trim 'md5:' prefix first)
-                if pg_pwd[3..] != *role.encrypted_password.as_ref().unwrap() {
+                // Check whether password changed or not (trim 'md5' prefix first if any)
+                let pg_pwd = match pg_pwd.strip_prefix("md5") {
+                    Some(s) => s.to_string(),
+                    None => pg_pwd.clone(),
+                };
+                if pg_pwd != *role.encrypted_password.as_ref().unwrap() {
```

Comment:
Hmm, so 'r.encrypted_password' doesn't include the 'md5' prefix, even if it's an MD5 hash? That's weird, but ok.

But hang on, `get_existing_roles` doesn't strip the 'md5' prefix, so I'm confused. If a role has an md5 hash, does `Role.encrypted_password` start with 'md5' or not?

Reply:
> But hang on, get_existing_roles doesn't strip the 'md5' prefix, so I'm confused. If a role has an md5 hash, does Role.encrypted_password start with 'md5' or not?

This is from the times when we were using md5 for everyone and hashes were stored in the console db without md5 prefix. So when role comes from the console (spec) `Role.encrypted_password` doesn't have md5 prefix, but when role comes from Postgres (`get_existing_roles`) it has this prefix.

I checked production and we still have a few dozens of pre-scram roles in the non-deleted projects, so looks like we still need a compatibility layer somewhere just to be on the safe side

Not sure what is a better place, but I like that `get_existing_roles` returns exactly what we have in the Postgres, so we do this md5 logic only in one place, when we do hashes comparison

Reply:
Please add a comment explaining that

Reply:
Sure, added a code comment with a bit rephrased explanation as above